src/argbash.m4

```diff
@@ -77,55 +72,56 @@ interpret_error()
-	_defines="${_defines}m4_define([OUTPUT_ABS_DIRNAME], [[$(cd "$_output_dirname" && pwd)]])"
-	printf "%s" "$_defines"
+	_output_dirname="$(dirname "${_intended_destination}")"
+	test "${_intended_destination}" != '-' && _defines="${_defines}m4_define([OUTPUT_BASENAME], [[$(basename "${_intended_destination}" "${SPURIONS_OUTPUT_SUFFIX}")]])"
+	_defines="${_defines}m4_define([OUTPUT_ABS_DIRNAME], [[$(cd "${_output_dirname}" && pwd)]])"
+	printf "%s" "${_defines}"
 }
 
-
+# Yes both tests need to pass or we should throw out an error message
+# shellcheck disable=2015
 assert_m4_files_are_readable()
 {
-	local _argbash_lib="$m4dir/argbash-lib.m4"
-	test -d "$m4dir" && test -x "$m4dir" || die "The directory '$m4dir' with files needed by Argbash is not browsable. Check whether it exists and review it's permissions."
-	test -r "$_argbash_lib" && test -f "$_argbash_lib" || die "The '$_argbash_lib' file needed by Argbash is not readable. Check whether it exists and review it's permissions."
-	test -r "$output_m4" && test -f "$output_m4" || die "The '$output_m4' file needed by Argbash is not readable. Check whether it exists and review it's permissions."
+	local _argbash_lib="${m4dir}/argbash-lib.m4"
+	test -d "${m4dir}" && test -x "${m4dir}" || die "The directory '${m4dir}' with files needed by Argbash is not browsable. Check whether it exists and review it's permissions."
+	test -r "${_argbash_lib}" && test -f "${_argbash_lib}" || die "The '${_argbash_lib}' file needed by Argbash is not readable. Check whether it exists and review it's permissions."
+	test -r "${output_m4}" && test -f "${output_m4}" || die "The '${output_m4}' file needed by Argbash is not readable. Check whether it exists and review it's permissions."
 }
 
-
 # The main function that generates the parsing script body
 # $1: The input file
 # $2: The output file
 # $3: The argument type
 do_stuff()
 {
-	local _pass_also="$_wrapped_defns" input prefix_len _ret
-	test "$_arg_commented" = on && _pass_also="${_pass_also}m4_define([COMMENT_OUTPUT])"
+	local _pass_also="${_wrapped_defns}" input prefix_len _ret
+	test "${_arg_commented}" = on && _pass_also="${_pass_also}m4_define([COMMENT_OUTPUT])"
 	_pass_also="${_pass_also}m4_define([_OUTPUT_TYPE], [[$3]])"
-	_pass_also="${_pass_also}$(define_file_metadata "$_arg_input" "$2")"
-	input="$(printf '%s\n' "$_pass_also" | cat - "$m4dir/argbash-lib.m4" "$output_m4")"
-	prefix_len=$(printf '%s\n' "$input" | wc -l)
-	input="$(printf '%s\n' "$input" | cat - "$1")"
-	run_autom4te "$input" 2> "$discard" \
-		| grep -v '^#\s*needed because of Argbash -->\s*$' \
-		| grep -v '^#\s*<-- needed because of Argbash\s*$'
+	_pass_also="${_pass_also}$(define_file_metadata "${_arg_input}" "$2")"
+	input="$(printf '%s\n' "${_pass_also}" | cat - "${m4dir}/argbash-lib.m4" "${output_m4}")"
+	prefix_len=$(printf '%s\n' "${input}" | wc -l)
+	input="$(printf '%s\n' "${input}" | cat - "$1")"
+	run_autom4te "${input}" 2> "${discard}" |
```

Comment:
here is the operator thing again.

Reply:
Ah now I see.  Ignore my earlier comment, I didn't think about the line continuation character.  Now that I have a better understanding I thing I can clean it up to your satisfaction using a regex in find and replace.